libusb/os/darwin_usb.c

```diff
@@ -654,12 +654,16 @@ static void darwin_exit (struct libusb_context *ctx) {
   if (0 == --init_count) {
     /* stop the event runloop and wait for the thread to terminate. */
     pthread_mutex_lock (&libusb_darwin_at_mutex);
-    CFRunLoopSourceSignal (libusb_darwin_acfls);
-    CFRunLoopWakeUp (libusb_darwin_acfl);
-    while (libusb_darwin_acfl)
-      pthread_cond_wait (&libusb_darwin_at_cond, &libusb_darwin_at_mutex);
-    pthread_mutex_unlock (&libusb_darwin_at_mutex);
-    pthread_join (libusb_darwin_at, NULL);
+    if (NULL != libusb_darwin_acfls && LIBUSB_DARWIN_STARTUP_FAILURE != libusb_darwin_acfls) {
```

Comment:
as far as I can tell - `LIBUSB_DARWIN_STARTUP_FAILURE == libusb_darwin_acfls` can never happen doe to:
```c
  if (libusb_darwin_acfl == LIBUSB_DARWIN_STARTUP_FAILURE) {
    libusb_darwin_acfl = NULL;
    rc = LIBUSB_ERROR_OTHER;
  }
```
inside `darwin_first_time_init`

surely it wouldn't hurt, but I don't see a code path that would lead to it
have I missed anything?

Reply:
Indeed. You are correct. Fixing.